### ISSUE TITLE
Questions about generate_latents_and_attribute_scores.py

### ISSUE THREAD

**Issue by NoctisZ** [yuval-alaluf/stylegan3-editing#41]
Great work and thank you so much for sharing the code! 
I got a couple of questions regarding the `generate_latents_and_attribute_scores.py` which I'm using to generate training data for new boundaries:
- What is the purpose of using `save_interval` in this function? Is there any reason that we don't want to save all generated data into a single folder (and a single scores.npy, ws.npy, etc.)? 
- I noticed the following description  of `save_interval` in README (for `generate_latents_and_attribute_scores.py` as well) -- "An npy file will be saved every save_interval samples". However, in the code Line 90, the if-statement `if seed_idx % save_interval == 0 and seed > 0` will only save a npy file at every (`save_interval `+ 1) step since seed_idx starts from 0. For example, if we set `n_images` to 8 and `save_interval` to 4, we want to have two npy files each containing 4 faces, but we'll end up getting only one npy file containing 5 faces. I think maybe this is a bug and need to fix it to something like `if (seed_idx + 1) % save_interval == 0 and seed > 0`? 

Would you kindly let me know if I'm missing something here? I would really appreciate it!

**Comment by woctezuma** [comment]
From the README:

> An `npy` file will be saved every `save_interval` samples. By default, we train boundaries for the aligned StyleGAN3
generator. If training boundaries for an unaligned generator, the psuedo-alignment trick will be performed before 
passing the images to the classifier.

In `generate_latents_and_attribute_scores.py`:

https://github.com/yuval-alaluf/stylegan3-editing/blob/e2a63fc7f072c4f8b8a7ba1dac8c7f87a8eb1647/editing/interfacegan/generate_latents_and_attribute_scores.py#L30-L31

https://github.com/yuval-alaluf/stylegan3-editing/blob/e2a63fc7f072c4f8b8a7ba1dac8c7f87a8eb1647/editing/interfacegan/generate_latents_and_attribute_scores.py#L90-L94

**Comment by NoctisZ** [comment]
I see your point, so basically with `if seed_idx % save_interval == 0 and seed > 0`, the first seed, indexed by `seed_idx=0`, will always be saved, and then starting from the second seed, we'll save them every `save_interval` step. 

I tried with `n_images` set to 5 and `save_interval` set to 2, and found that data in _id_0_ folder contains 3 faces since the first seed is saved together with the next two, and the data in _id_1_ folder contains 2 faces. However, if `n_images` is set to 4 and `save_interval` is 2, I can only get one folder (ie. _id_0_) that contains 3 faces and lost data for the 4th face. I guess I just need to be careful with this hyperparameter setting then.

**Comment by woctezuma** [comment]
Yes, I don't think it is a bug, and one needs to be careful with this tricky parameter.

**Comment by NoctisZ** [comment]
Yes totally agree. Thanks for the response!
